### DISCUSSION TITLE
Add Event map support (world info, enemy context, edges, etc.)

### DISCUSSION THREAD

**Discussion by Danielosama** [XVs32/kcauto_custom#58] [Q&A] [answered]
Hello,

I am looking to add Event map support (I am still using Perry's original KCAuto, still works fine for me).

I know where to get edges info from: https://github.com/KC3Kai/KC3Kai/blob/master/src/data/edges.json

However, I am unsure as to where all the rest of the info comes from (see screen capture below):

![image](https://github.com/XVs32/kcauto_custom/assets/12687427/8aabb7ef-5ef1-423e-a7b2-2d89831eb863)

Is any of it automated?

Most of it is self-explanatory (world, subworld, panel, and page); I'm guessing the coords have to be manually typed?

And my biggest doubt is what is "enemy_context" exactly?

Thank you in advance.

**Comment by XVs32** [comment, marked as answer]
Hello [Danielosama](https://github.com/Danielosama),

Thanks for the question.

---

## What is `enemy_context`?

`enemy_context` is a tag which tells kcauto what kind of enemy you could find in this map.
That way the quest module can select the corresponding quest for this map.

For example, with the tag `[carriers, subs]` here,  
the quest module will select quests like `Bd4	敵空母を3隻撃沈せよ！` and `Bd8	敵潜水艦を制圧せよ！`

---

## How do you get coords?

Unfortunately, yes. You do have to find the coords manually.

The good news is, however, I did make a simple tool that helps the process.
It is [get_mouse_location.py](https://github.com/XVs32/kcauto_custom/blob/develop/tool/get_mouse_location.py) located in `/tool`.
You click the top left corner of the game window first, then click on each node on the map, this tool will give you the coords you need.

---

Sincerely,
XVs32

**Reply by Danielosama** [reply]
Thank you very much for the detailed answers.

Forgive me but I am not really proficient with Python, how would I use [get_mouse_location.py](https://github.com/XVs32/kcauto_custom/blob/develop/tool/get_mouse_location.py) ?
Just double click it and follow the instructions you provided?

I'll try to set up E1 support and see how it goes.

On an unrelated note, I was thinking of doing a step-by-step installation guide of KCAuto (Perry's original one, although it seems the steps are similar to your fork) in PDF and video format if that is alright with you.

All best,

**Reply by XVs32** [reply]
Hello [Danielosama](https://github.com/Danielosama),

---

| Just double click it and follow the instructions you provided?

Sadly you can't simply double click a `.py` file to run it.
Instead, you will have to run it under python.
Here is a tutorial https://youtu.be/s8bp1OavluQ?t=214

If you are familiar with the original kcauto, you should have python installed on your PC already.

---

| doing a step-by-step installation guide

Sure! I appreciate you asking my permission but you don't have to.
The environment setup is always frustrating, it helps a lot if we have a step-by-step installation guide.

---

Sincerely,
XVs32
